src/index.js

```diff
@@ -0,0 +1,49 @@
+import Plugin from 'broccoli-caching-writer';
+import FaviconsJs from 'favicons';
+
+import deepMerge from 'lodash.merge';
+import path from 'path';
+import fs from 'fs';
+
+import config from './config';
+
+export default class Favicon extends Plugin {
+  constructor(node, options = {}) {
+    super([node], {
+      name: options.name,
+      annotation: options.annotation
+    });
+
+    this.config = deepMerge({}, config, options);
+  }
+
+  build() {
+    let { iconPath, onSuccess, faviconsConfig } = this.config;
+    iconPath = path.join(this.inputPaths[0], iconPath);
+
+    if (!fs.existsSync(iconPath)) {
+      return Promise.resolve()
```

Comment:
Should we print out a warning of some kind here? I could imagine someone getting confused about where to put the source file (i.e. they drop in the wrong folder) and then getting frustrated by the fact that the addon appears to not be working. Since all of this is opt-in, I can't imagine a scenario where you'd legitimately want to have this addon installed, but not have the source icon file exist.

Reply:
Well, there is a case. Eg you want to test out PWA thanks to generated manifests and add the favicon later.

But sure. That would be appreciated to add a warning like: `[WARRNING] Favicon file has been not detected in path "favcon/path/file.png"`
I will add that!